daemons/pid/simple.py

```diff
@@ -32,12 +33,31 @@ def pid(self):
 
                 try:
 
-                    return int(pidfile.read().strip())
+                    pid = int(pidfile.read().strip())
 
                 except ValueError:
 
                     return None
 
+		try:
+
+		    os.kill(pid, 0)
+
+		except OSError as e:
+
+		    if e.errno == errno.EPERM:
+
+			return pid
```

Comment:
Hi Kevin,
there are a few ways you can get here, but I'm not convinced there's a 100% correct solution within the daemons code itself.
If the pid file is stale because the OS crashed and you're unlucky and the pid in the file now matches a running process that has nothing to do with you, that's a problem. The only correct way to handle that is to make sure you clean up stale pid files at boot (which is why a lot of system daemons place the file in /var/run so it can be cleaned up.)

The other option however is when the daemon has been started but under a different uid. Normally, I don't think you're going to get here because the open for write of the pidfile is going to fail. If that does succeed, the rationale is that because the caller decides to try to run the daemon based on whether we return a valid pid or None, if we return None, it will try to run a second instance of the daemon which is the wrong thing to do. So returning the pid prevents the caller from erroneously trying to start the daemon twice.